### ISSUE TITLE
Playlist Block: Accessibility of compound labels

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#75584]
### Description

The buttons to trigger and navigate playlist items wrap the entirety of the visible text. That's not a problem, but it results in the text being run together awkwardly. The file length has a screen-reader-text wrapper with 'Length:', and the colon triggers a pause in screen readers. The pause is great; but the lack of pauses elsewhere is awkward. 

The reading is "number title album Length: [pause] length Select to play this track"

This would be more natural if there were a few additional pauses. I'd recommend that pauses be inserted between each composite section. This could be messy in code, but I think the structure would be better. 

This works: https://codepen.io/joedolson/pen/YPWbWgR

### Step-by-step reproduction instructions

1. Insert a playlist block with tracks.
2. Enable your screen reader.
3. Navigate to an item and observe that the text is read with no breaks, obscuring meaning.

### Environment info

Playground with PR https://github.com/WordPress/gutenberg/pull/75203

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
